tests/packages/test_directory_dependency.py

```diff
@@ -75,3 +75,14 @@ def test_directory_dependency_pep_508_extras() -> None:
     requirement = f"demo[foo,bar] @ file://{path.as_posix()}"
     requirement_expected = f"demo[bar,foo] @ file://{path.as_posix()}"
     _test_directory_dependency_pep_508("demo", path, requirement, requirement_expected)
+
+
+def test_directory_dependency_pep_517_non_poetry() -> None:
+    path = Path(__file__).parent.parent / "fixtures" / "project_with_pep517_non_poetry"
+
+    try:
+        DirectoryDependency("flit", path)
+    except ValueError as e:
```

Comment:
when would this be expected to raise?

Reply:
If the detection of build system for the dir dep is broken, the initialisation will raise a value error.